modules/cover-image-workflowoperation/src/main/java/org/opencastproject/workflow/handler/coverimage/CoverImageWorkflowOperationHandlerBase.java

```diff
@@ -302,57 +304,87 @@ protected String loadXsl(WorkflowOperationInstance operation) throws WorkflowOpe
     }
   }
 
-  protected String getMetadataXml(MediaPackage mp) {
-    StaticMetadata metadata = getStaticMetadataService().getMetadata(mp);
+  protected String getMetadataXml(MediaPackage mp, WorkflowOperationInstance operation) {
+    //get specified episode/series flavor
+    String configuredEpisodeFlavor = StringUtils.trimToEmpty(operation.getConfiguration(EPISODE_FLAVOR));
```

Comment:
This is easier than to check manually for empty values:
```suggestion
    final String configuredEpisodeFlavor = Objects.toString(StringUtils.trimToNull(operation.getConfiguration(EPISODE_FLAVOR)), "dublincore/episode);
```